### DISCUSSION TITLE
eval_model function results differ from test_step function.

### DISCUSSION THREAD

**Discussion by SailSabnis** [mrdbourke/pytorch-deep-learning#1040] [Q&A] [answered]
Hi all, 

From 03. PyTorch Computer Vision where we create a CNN model for FashionMNIST data - 

We first train and test the model using the train_step and test_step function. I understand we later run it for two epochs to begin with to get the test loss and test accuracy. After that we run the eval_model function to capture model loss and accuracy in a variable. In the video, Daniel gets the model loss and accuracy (i.e. eval_model results) same as the test_step function. But I get different results. Why is that so. What am I missing? 

**Screenshot 1 - My results** 
<img width="409" alt="Screenshot 2024-08-14 at 5 47 47 PM" src="https://github.com/user-attachments/assets/33807165-ec79-40dc-93f7-85272430e231">

**Daniels results from the video 
<img width="522" alt="Screenshot 2024-08-14 at 5 54 01 PM" src="https://github.com/user-attachments/assets/3b84f763-efc4-414e-90b3-4056edbcdbe0">
<img width="826" alt="Screenshot 2024-08-14 at 5 54 45 PM" src="https://github.com/user-attachments/assets/ce4c841a-6da1-456d-9e7f-176015534b01">

**Comment by SailSabnis** [comment, marked as answer]
FIgured it out. In my eval_model function - 

model_loss /= len(dataloader)
model_acc /= len(dataloader)

should be outside the for loop.

**Reply by mrdbourke** [reply]
Glad you figured it out!

Those kind of errors get me too!
